### ISSUE TITLE
a11y audit: MultiSelect a11y fixes

### ISSUE THREAD

**Issue by marijohannessen** [IBM/carbon-components-react#615]
### Detailed description

- [x] All content must reside within a WAI-ARIA landmark or labelled region role.
<img width="996" alt="screen shot 2018-02-15 at 11 20 07 am" src="https://user-images.githubusercontent.com/5447411/36270876-8483de58-1242-11e8-95b0-6cbc619622a4.png">

- [x] The id "identifier" referenced by the 'for' attribute of a label element is not a valid form input element or an element with id "identifier" is either null or does not exist.
<img width="1008" alt="screen shot 2018-02-15 at 11 19 55 am" src="https://user-images.githubusercontent.com/5447411/36270794-48c14eb4-1242-11e8-81d8-1c5ba8e3a1c0.png">

- [x] The identifier element has the id "identifier" that is either empty or already in use.
<img width="984" alt="screen shot 2018-02-15 at 11 19 43 am" src="https://user-images.githubusercontent.com/5447411/36270795-48d92fde-1242-11e8-82f6-87e19b115e0d.png">

**Comment by tw15egan** [comment]
@joshblack it seems like the `id` is being set on both the wrapping item div, as well as the checkbox. When I try to change one of them to a different value, it breaks. Any ideas on how we can avoid using duplicate ids here? 

![screen shot 2018-03-02 at 1 09 29 pm](https://user-images.githubusercontent.com/11928039/36916931-fde86096-1e1a-11e8-83cc-7db6d70763ac.png)

**Comment by joshblack** [comment]
@tw15egan I think we could do two things, either grab the id from `itemProps` before spreading it onto `ListBox.MenuItem`, or just append `-input` to the id in `Checkbox`. Do you think one of those options could work?

```js
                      const { id, ...itemProps } = getItemProps({ item });
                      const itemText = itemToString(item);
                      const isChecked = selectedItem.indexOf(item) !== -1;
                      return (
                        <ListBox.MenuItem
                          key={itemProps.id}
                          isActive={selectedItem.indexOf(item) !== -1}
                          isHighlighted={highlightedIndex === index}
                          {...itemProps}>
                          <Checkbox
                            id={id}
                            name={itemText}
                            checked={isChecked}
                            readOnly={true}
                            tabIndex="-1"
                            labelText={itemText}
                          />
                        </ListBox.MenuItem>
                      );
```

Or:

```js
                      const itemProps = getItemProps({ item });
                      const itemText = itemToString(item);
                      const isChecked = selectedItem.indexOf(item) !== -1;
                      return (
                        <ListBox.MenuItem
                          key={itemProps.id}
                          isActive={selectedItem.indexOf(item) !== -1}
                          isHighlighted={highlightedIndex === index}
                          {...itemProps}>
                          <Checkbox
                            id={`${itemProps.id}-input`}
                            name={itemText}
                            checked={isChecked}
                            readOnly={true}
                            tabIndex="-1"
                            labelText={itemText}
                          />
                        </ListBox.MenuItem>
                      );
```

**Comment by tw15egan** [comment]
Hmm I tried the latter first, and it ended up selecting multiple / no checkboxes. The other implementation seems to do the same. I'm not too familiar with how this component works but it seems like the selection + actions are tied to the `id`?

**Comment by joshblack** [comment]
Ooof yeah I see what you're saying. Could we just add an `onClick` to `Checkbox` that stops the event from propagating?

**Comment by tw15egan** [comment]
@joshblack that did the trick! 🎉
